### PR TITLE
Allow multiple IDs

### PR DESCRIPTION
**Pull request by eivindhammers**
Allow for multiple entry IDs, as is implied by the description of "entryid" in docs ("The entry ID(s)."). Also removed the line requesting round info, which doesn't seem to be used.

### CODE REVIEW

**Comment by eivindhammers** [comment]
I also added a function to retrieve captains (for multiple entries/GWs). Feel free to drop that from the pull request if it is of limited interest.

File: R/entry_functions.R [hunk 1]
```diff
@@ -97,3 +97,30 @@ get_entry_picks <- function(entryid = NULL, gw = NULL){
   return(picks2)
 }
 
+get_entry_captain <- function(entryid = NULL, gw = NULL, player_names = FALSE){
```
Comment (wiscostret) [inline R/entry_functions.R]: I would remove and default the player_names functionality - it´s a nice touch, and should be useful in the vast majority of cases; the get_player_name() function is light enough that this should not add too much to the load.
```suggestion
get_entry_captain <- function(entryid = NULL, gw = NULL){
```

File: R/entry_functions.R [hunk 2]
```diff
@@ -97,3 +97,30 @@ get_entry_picks <- function(entryid = NULL, gw = NULL){
...
+      picks <- jsonlite::fromJSON(paste("https://fantasy.premierleague.com/api/entry/",entryid[j],"/event/",gw[i],"/picks","/",sep=""))
+      captain <- data.frame("id" = picks$picks[picks$picks$multiplier %in% c(2, 3), "element"])
+      captain$entry <- entryid[j]
+      captain2 <- rbind(captain2, captain)
+    }
+    captain2$event <- gw[i]
+    captain3 <- rbind(captain3, captain2)
+  }
+  
+  if (player_names == TRUE) {
```
Comment (wiscostret) [inline R/entry_functions.R]: Cf. above, remove the if() statement here.
```suggestion
```

File: R/entry_functions.R [hunk 3]
```diff
@@ -97,3 +97,30 @@ get_entry_picks <- function(entryid = NULL, gw = NULL){
...
+    player_names <- get_player_name(unique(captain3$captain))
+    captain3 <- merge(captain3, player_names, by = "id")
+    captain3 <- captain3[c(1, 4, 2, 3)]
+    names(captain3) <- c("captain_id", "captain_name", "entry", "event")
+  }
```
Comment (wiscostret) [inline R/entry_functions.R]: Cf. above, remove the if() statement wrapper here.

Would also add a reordering here to sort the return by entry and event
```suggestion
captain3 <- captain3[order(captain3$entry,captain3$event),]
```

File: R/entry_functions.R [hunk 4]
```diff
@@ -97,3 +97,30 @@ get_entry_picks <- function(entryid = NULL, gw = NULL){
...
+    }
+    captain2$event <- gw[i]
+    captain3 <- rbind(captain3, captain2)
+  }
+  
+  if (player_names == TRUE) {
+    player_names <- get_player_name(unique(captain3$captain))
+    captain3 <- merge(captain3, player_names, by = "id")
+    captain3 <- captain3[c(1, 4, 2, 3)]
+    names(captain3) <- c("captain_id", "captain_name", "entry", "event")
```
Comment (wiscostret) [inline R/entry_functions.R]: Renaming these to fit with package norms, allowing for easier merges with other function results.
```suggestion
    names(captain3) <- c("id", "playername", "entry", "event")
```

**Review by wiscostret — CHANGES_REQUESTED** [review]
Many thanks for this - this is all good ideas. If you want, you can separate the captains commit out in a separate pull request - I am happy to approve this with the requested changes. I am more in doubt about the other commit (allowing multiple entry IDs), but I will comment in detail on that.

**Review by wiscostret — COMMENTED** [review]
Thanks for this also. Good spot on the roundinfo, not sure how that´s ended up there, poor quality checking on my part.

On adding functionality for multiple entry IDs, I´ve considered this a lot before. I shared some thoughts in a prior issue [here](https://github.com/wiscostret/fplscrapR/issues/15). The get_entry_picks() function returns a list of lists (and actually, in part, a list of list of lists), because of how the FPL API is (unhelpfully) structured. Adding multiple IDs with the suggested changes returns another list layer, so effectively you get a list of list of list of lists (phew). This is not a user-friendly data structure to work with, which is why I´ve preferred not to add this option, but I´m very open to further considerations.

As I mentioned on the issue, an alternative would be to add more specifically targeted functions, like a get_entry_player_picks() which would return only the $picks dataframe (which is what most users are looking for). This would be easier to add a multiple entry ID functionality to. Any further comments welcome.

**Comment by eivindhammers** [comment]
Thanks for your feedback. The captains function is now in its own pull request, with suggested changes.

I second the idea of a `get_entry_player_picks()` function, returning a dataframe. The nested list returned by my function is something of a mess, yes.

**Comment by eivindhammers** [comment]
I guess this PR can be closed. I might open a new one with a `get_entry_player_picks()` function.

**Comment by eivindhammers** [comment]
Closing this PR, see #17 and #18.

**Pull request closed without merging**